x-pack/platform/plugins/shared/actions/docs/connector_spec/connector_spec.ts

```diff
@@ -0,0 +1,283 @@
+import type { AxiosInstance } from 'axios';
+import { i18n } from '@kbn/i18n';
+import { withUIMeta, UISchemas } from './connector_spec_ui';
+
+export { withUIMeta, UISchemas } from './connector_spec_ui';
+
+// ============================================================================
+// INTERNATIONALIZATION
+// ============================================================================
+
+export function createI18nKeys(connectorId: string) {
+  const base = `xpack.stackConnectors${connectorId}`;
+  return {
+    metadata: (key: string) => `${base}.metadata.${key}`,
+    config: (key: string) => `${base}.config.${key}`,
+    secrets: (key: string) => `${base}.secrets.${key}`,
+    actions: (actionName: string, key: string) => `${base}.actions.${actionName}.${key}`,
+    validation: (key: string) => `${base}.validation.${key}`,
+    test: (key: string) => `${base}.test.${key}`,
+  };
+}
+
+// ============================================================================
+// METADATA
+// ============================================================================
+
+export interface ConnectorMetadata {
+  id: string;
+  displayName: string;
+  icon?: string;
+  description: string;
+  docsUrl?: string;
+  minimumLicense: 'basic' | 'gold' | 'platinum' | 'enterprise';
+  supportedFeatureIds: Array<
+    | 'alerting'
+    | 'cases'
+    | 'uptime'
+    | 'security'
+    | 'siem'
+    | 'generativeAIForSecurity'
+    | 'generativeAIForObservability'
+    | 'generativeAIForSearchPlayground'
+    | 'endpointSecurity'
+  >;
+}
+
+// ============================================================================
+// STANDARD AUTH SCHEMAS
+// ============================================================================
+
+export const BasicAuthSchema = z.object({
+  username: z.string().describe('Username'),
+  password: withUIMeta(z.string(), { sensitive: true }).describe('Password'),
+});
+
+export const BearerAuthSchema = z.object({
+  apiKey: withUIMeta(z.string(), { sensitive: true }).describe('API Key'),
+});
+
+export const OAuth2AuthSchema = z.object({
+  clientId: z.string().describe('Client ID'),
+  clientSecret: withUIMeta(z.string(), { sensitive: true }).describe('Client Secret'),
+  tokenUrl: z.string().url().describe('Token URL'),
+  scope: z.string().optional().describe('Scope'),
+});
+
+export const SSLAuthSchema = z.object({
+  certificateType: z.enum(['crt', 'pfx']).describe('Certificate Type'),
+  certificate: withUIMeta(z.string(), { sensitive: true }).optional().describe('Certificate'),
+  privateKey: withUIMeta(z.string(), { sensitive: true }).optional().describe('Private Key'),
+  pfx: withUIMeta(z.string(), { sensitive: true }).optional().describe('PFX Bundle'),
+  passphrase: withUIMeta(z.string(), { sensitive: true }).optional().describe('Passphrase'),
+  ca: z.string().optional().describe('CA Certificate'),
+});
+
+export function getAuthSchema(types: Array<'basic' | 'bearer' | 'oauth2' | 'ssl'>) {
```

Comment:
In this concept, how will this be called and by whom?

How about having `Array<'basic' | 'bearer' | 'oauth2' | 'ssl'>` be an actual field of `SingleFileConnectorDefinition`, something like `authTypes`, and then registering the connector in the framework handles this transformation to the actual schema fields?